calico_containers/calico_ctl/checksystem.py

```diff
@@ -177,3 +181,36 @@ def _check_docker_version(libnetwork=False):
             system_ok = False
 
     return system_ok
+
+def _check_etcd_version():
+    """
+    Check that etcd is running and supported.
+
+    :return: True if etcd version is OK.
+    """
+    system_ok = True
+
+    minimum_version_string = ETCD_VERSION
+    minimum_version = normalize_version(minimum_version_string)
+
+    try:
+        detected_version_string = \
+            client.etcd_client.api_execute("/version", 'GET').data
+    except EtcdConnectionFailed:
+        print >> sys.stderr, "ERROR: Could not connect to etcd at %s" % \
+            etcd_authority
+        system_ok = False
+    else:
+        # Remove everything except numbers and dots from the string.
+        detected_version_string = re.sub(r'[^\d\.]', '',
```

Comment:
This isn't normal semantic versioning is it?

Normally you'd have x.y.z with a possible extra bit at the end separated by a hyphen (e.g.1.0.0-alpha.1).

In the example above the tuple would end up as (1, 0, 1) which is incorrect.